spec/mauth_client_spec.rb

```diff
@@ -52,231 +66,520 @@ def (::Rails).logger
 
     it 'initializes with app_uuid' do
       uuid = "40e19273-6a43-41d1-ba71-71cbb1b69d35"
-      [{:app_uuid => uuid}, {'app_uuid' => uuid}].each do |config|
+      [{app_uuid: uuid}, {'app_uuid' => uuid}].each do |config|
         mc = MAuth::Client.new(config)
         expect(uuid).to eq(mc.client_app_uuid)
       end
     end
 
     it 'initializes with ssl_cert_path' do
       ssl_certs_path = 'ssl/certs/path'
-      [{:ssl_certs_path => ssl_certs_path}, {'ssl_certs_path' => ssl_certs_path}].each do |config|
+      [{ssl_certs_path: ssl_certs_path}, {'ssl_certs_path' => ssl_certs_path}].each do |config|
         mc = MAuth::Client.new(config)
         expect(ssl_certs_path).to eq(mc.ssl_certs_path)
       end
     end
 
     it 'initializes with private key' do
       key = OpenSSL::PKey::RSA.generate(2048)
-      [{:private_key => key}, {'private_key' => key}, {:private_key => key.to_s}, {'private_key' => key.to_s}].each do |config|
+      [{private_key: key}, {'private_key' => key}, {private_key: key.to_s}, {'private_key' => key.to_s}].each do |config|
         mc = MAuth::Client.new(config)
         # can't directly compare the OpenSSL::PKey::RSA instances
         expect(key.class).to eq(mc.private_key.class)
         expect(key.to_s).to eq(mc.private_key.to_s)
       end
     end
+
+    it 'correctly initializes with authenticate_with_only_v2 as true with boolean true or string "true"' do
+      [true, 'true'].each do |authenticate_with_only_v2|
+        [{:authenticate_with_only_v2 => authenticate_with_only_v2}, {'authenticate_with_only_v2' => authenticate_with_only_v2}].each do |config|
```

Comment:
I realize there's a lot of existing style violations in this file, but for new changes, could you double check the style? For example:
`{:authenticate_with_only_v2 => authenticate_with_only_v2}`
`{ authenticate_with_only_v2: authenticate_with_only_v2 }`